src/ccache.cpp

```diff
@@ -1272,17 +1272,24 @@ hash_common_info(const Context& ctx,
   hash.hash(Util::base_name(args[0]));
 
   // Hash variables that may affect the compilation.
-  const char* always_hash_env_vars[] = {
+  struct
+  {
+    const char* name;
+    uint32_t sloppiness;
+  } hash_env_vars[] = {
     // From <https://gcc.gnu.org/onlinedocs/gcc/Environment-Variables.html>:
-    "COMPILER_PATH",
-    "GCC_COMPARE_DEBUG",
-    "GCC_EXEC_PREFIX",
-    "SOURCE_DATE_EPOCH",
+    {"COMPILER_PATH", 0},
+    {"GCC_COMPARE_DEBUG", 0},
+    {"GCC_EXEC_PREFIX", 0},
+    {"SOURCE_DATE_EPOCH", SLOPPY_SOURCE_DATE_EPOCH},
   };
-  for (const char* name : always_hash_env_vars) {
-    const char* value = getenv(name);
+  for (const auto& env : hash_env_vars) {
+    if (env.sloppiness && (ctx.config.sloppiness() & env.sloppiness)) {
+      continue;
```

Comment:
Why not cache it always? The others are also always cached.
This invalidates otherwise valid cache entries if sloppiness is added after the first run?!

Reply:
>Why not cache it always? The others are also always cached.

debhelpers passed it unconditionally, using the last entry from debian/changelog, or current time if there is no entries.

IIUC it is not set if there is already such env variable, however, it does not sounds right to set this value to 0 (or similar), since this will affect `__TIME__`/`__DATE__`, while ccache will not cache those translation units anyway (if `time_macros` sloppiness is not set)

>This invalidates otherwise valid cache entries if sloppiness is added after the first run?!

Yes, like other sloppiness, i.e. `locale`